src/amber/cli/commands/encrypt.cr

```diff
@@ -0,0 +1,41 @@
+require "../../support/message_encryptor"
+
+module Amber::CLI
+  class MainCommand < ::Cli::Supercommand
+    command "e", aliased: "encrypt"
+
+    class Encrypt < ::Cli::Command
+      class Options
+        arg "env", desc: "environment file to encrypt", default: "production"
+        string ["-e", "--editor"], desc: "editor", default: "vim"
+      end
+
+      class Help
+        caption "# Encrypts Environment"
+      end
+
+      def run
+        secret_key = ENV["AMBER_SECRET_KEY"]? || File.open(".amber_secret_key").gets_to_end.to_s
+        env = ENV["AMBER_ENV"]? || args.env
+        encrypted_file = "config/environments/.#{env}.enc"
+        unencrypted_file = "config/environments/#{env}.yml"
+
+        enc = Amber::Support::MessageEncryptor.new(secret_key.to_slice)
+
+        if File.exists?(encrypted_file)
+          File.write(unencrypted_file, enc.decrypt(File.open(encrypted_file).gets_to_end.to_slice))
+          system("#{options.editor} #{unencrypted_file}")
+        end
+
+        if File.exists?(unencrypted_file)
+          File.write(encrypted_file, enc.encrypt(File.read(unencrypted_file)))
+          File.delete(unencrypted_file)
+        else
+          puts "Ooops! Your environment file doesn't exist!"
+        end
+      rescue
+        puts "Failed! Are you sure your key is correct?"
```

Comment:
Rails uses secret_key_base and then bases other keys off of it. So that they'll stay the same every-time the project runs but every key in the system can still be different because they use secret_key_base and as seed.

Reply:
Perhaps although the .encryption_key for secrets is different that secret_key_base.

Reply:
When I read this error message I ask myself which secret key? I think this should be more specific